parser/src/powdr.lalrpop

```diff
@@ -152,6 +152,9 @@ Machine: Machine<T> = {
 }
 
 MachineArguments: MachineArguments = {
+    "(" "_" "," "_" ")" => MachineArguments::default(),
+    "(" <latch:Identifier> "," "_" ")" => MachineArguments { latch: Some(latch), function_id: None },
+    "(" "_" "," <function_id:Identifier> ")" => MachineArguments { latch: None, function_id: Some(function_id) },
```

Comment:
Read-only registers are cleaner because of this bit in `vm_to_constrained`
```rust
                          // Unconstrain read-only registers when calling `_reset`
                          ReadOnly => {
                              let not_reset =
                                  build_sub(build_number(1u64), direct_reference("instr__reset"));
                              (build_mul(not_reset.clone(), lhs), build_mul(not_reset, rhs))
                          }
```